containers/src/Data/IntMap/Merge/Strict.hs

```diff
@@ -6,15 +6,9 @@
 #if !defined(TESTING) && defined(__GLASGOW_HASKELL__)
 {-# LANGUAGE Trustworthy #-}
 #endif
-#if __GLASGOW_HASKELL__ >= 708
 {-# LANGUAGE RoleAnnotations #-}
 {-# LANGUAGE TypeFamilies #-}
-#define USE_MAGIC_PROXY 1
-#endif
-
-#if USE_MAGIC_PROXY
 {-# LANGUAGE MagicHash #-}
-#endif
```

Comment:
Shouldn't this be hidden behind an `#ifdef __GLASGOW_HASKELL__`?

Reply:
Do you know any non-GHC compiler I can use to test this?

Reply:
No, and that troubles me greatly. There's tradition here, long preceding my maintainership, of trying to maintain portability, but it gets harder and harder.

Reply:
Hmm I have to revert most of the changes because of coercions then. TBH I'm skeptical about the utility of untested code for non-GHC...

Reply:
I share your skepticism. Should we take it to the libraries list?

Reply:
Yes please!

Reply:
Would you like to start it off, or shall I?

Reply:
Can you start it off? I don't know much about the history nor the policy so...

Reply:
Nevermind, I'll go ahead an post a mail

Reply:
Thanks!